src/test/java/com/segment/analytics/android/integrations/adjust/AdjustIntegrationTest.java

```diff
@@ -160,6 +160,55 @@
     verify(adjustInstance).trackEvent(event);
   }
 
+  @Test public void trackWithAnonymousId() throws Exception {
+    AdjustEvent event = mock(AdjustEvent.class);
+    PowerMockito.whenNew(AdjustEvent.class).withArguments("bar").thenReturn(event);
+
+    Traits traits = createTraits() //
+        .putValue("anonymousId", "1234");
+
+    integration.identify(new IdentifyPayloadBuilder().traits(traits).build());
+    integration.track(new TrackPayloadBuilder()
+        .event("foo")
+        .build());
+    verify(adjustInstance).trackEvent(event);
+    verify(adjustInstance).addSessionPartnerParameter("anonymousId", "1234");
+
+  }
+
+  @Test public void trackWithUserId() throws Exception {
+    AdjustEvent event = mock(AdjustEvent.class);
+    PowerMockito.whenNew(AdjustEvent.class).withArguments("bar").thenReturn(event);
+
+    Traits traits = createTraits() //
+        .putValue("userId", "5678");
+
+    integration.identify(new IdentifyPayloadBuilder().traits(traits).build());
+    integration.track(new TrackPayloadBuilder()
+        .event("foo")
+        .build());
+    verify(adjustInstance).trackEvent(event);
+    verify(adjustInstance).addSessionPartnerParameter("userId", "5678");
+
+  }
+
+  @Test public void trackWithBoth() throws Exception {
```

Comment:
I don't think this is a great name. `trackWithUserIdAndAnonymousId` is more descriptive.